pennylane/qnode.py

```diff
@@ -499,6 +482,23 @@ def evaluate(self, args, **kwargs):
         Variable.kwarg_values = keyword_values
 
         self.device.reset()
+
+        # check that no wires are measured more than once
+        m_wires = list(w for ex in self.ev for w in ex.wires)
+        if len(m_wires) != len(set(m_wires)):
+            raise QuantumFunctionError('Each wire in the quantum circuit can only be measured once.')
+
+        def check_op(op):
+            """Make sure only existing wires are referenced."""
+            for w in op.wires:
+                if w < 0 or w >= self.num_wires:
+                    raise QuantumFunctionError("Operation {} applied to wire {}, "
+                                               "device only has {}.".format(op.name, w, self.num_wires))
+
+        # check every gate/preparation and ev measurement
+        for op in self.ops:
+            check_op(op)
+
```

Comment:
Since `wires` can now be `Variables`, all validation must be moved from `qnode.construct` (since keyword arguments are unknown on construction) to `qnode.evaluate` (where they will be known).